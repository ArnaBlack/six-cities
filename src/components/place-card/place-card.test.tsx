import * as React from 'react';
import {
  configure,
  shallow,
} from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import PlaceCard from './place-card';

configure({adapter: new Adapter()});

const mock = {
  offer: {
    bedrooms: 1,
    city: {
      name: `Brussels`,
      location: {
        latitude: 50.846557,
        longitude: 4.351697,
        zoom: 13,
      },
    },
    description: `Description`,
    goods: [`Washer`, `Towels`],
    host: {
      avatarUrl: `img/avatar-angelina.jpg`,
      id: 25,
      isPro: true,
      name: `Name`,
    },
    id: 1,
    images: [`https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`],
    isFavorite: true,
    isPremium: false,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13,
    },
    maxAdults: 3,
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
    price: 117,
    rating: 3.4,
    title: `The house among olive`,
    type: `room`,
  },
};

it(`PlaceCard correctly renders`, () => {
  const {offer} = mock;
  const clickHandler = jest.fn();
  const tree = shallow(<PlaceCard
    offer={offer}
    onImageClick={clickHandler}
  />);

  expect(tree).toMatchSnapshot();
});
