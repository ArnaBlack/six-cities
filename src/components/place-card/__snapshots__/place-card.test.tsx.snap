// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard correctly renders 1`] = `
<article
  className="place-card undefined"
>
  <div
    className="place-card__image-wrapper undefined"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      <img
        alt="Place image"
        className="place-card__image"
        height="200"
        src="https://es31-server.appspot.com/six-cities/static/hotel/4.jpg"
        width="260"
      />
    </a>
  </div>
  <div
    className="place-card__info undefined"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          117
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <ConnectFunction
        className="place-card"
        height={19}
        id={1}
        isFavorite={true}
        width={18}
      />
    </div>
    <Rating
      hasValue={false}
      rating={3.4}
      ratingClass="place-card__rating"
      starsClass="place-card__stars "
    />
    <h2
      className="place-card__name"
    >
      <Link
        to="/offer/1"
      >
        The house among olive
      </Link>
    </h2>
    <p
      className="place-card__type"
    >
      Private Room
    </p>
  </div>
</article>
`;
